### PR TITLE
Add support for CosmosDB virtual network rules and AKS RBAC.

### PR DESCRIPTION
**Pull request by pgavlin**
Just what it says on the tin.

Concretely, the former involves adding two properties to `azurerm_cosmosdb_account`--`is_virtual_network_filter_enabled` and `virtual_network_rules`--that correspond to the similarly-named REST API parameters. The latter involves adding a single boolean property, `enable_rbac`, to `azurerm_kubernetes_cluster`, that also corresponds to a similarly-named REST API parameter.

### CODE REVIEW

**Comment by lukehoban** [comment]
See https://github.com/terraform-providers/terraform-provider-azurerm/pull/1820 for upstream work on RBAC in AKS.  It includes some additional things that are not here which may be required - `ForceNew: true` and some different KubeConfig parsing logic.  Will want to review that and potentially merge that in early here instead.

The CosmosDB change has no upstream PR but is tracked by https://github.com/terraform-providers/terraform-provider-azurerm/issues/1342.

File: azurerm/resource_arm_cosmos_db_account.go [hunk 1]
```diff
@@ -205,6 +205,26 @@ func resourceArmCosmosDBAccount() *schema.Resource {
 				Set: resourceAzureRMCosmosDBAccountCapabilitiesHash,
 			},
 
+			"is_virtual_network_filter_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
```
Comment (lukehoban) [inline azurerm/resource_arm_cosmos_db_account.go]: Does this (and all others) need `ForceNew`?

Comment (pgavlin) [reply azurerm/resource_arm_cosmos_db_account.go]: Nope, these are updatable in-place.

**Comment by pgavlin** [comment]
@lukehoban this is ready to go. I've verified that the RBAC changes do enable RBAC, and I've aligned the schema with the changes in terraform-providers/terraform-provider-azurerm#1820.

**Pull request merged**